xip/xipether.c

```diff
@@ -0,0 +1,43 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include <sys/types.h>
+#include <fcntl.h>
+#include <asm/byteorder.h>
+#include <asm-generic/errno-base.h>
+#include <net/xia_fib.h>
+#include <xia_socket.h>
+#include <sys/socket.h>
+
+#include "xip_common.h"
+#include "utils.h"
+#include "ll_map.h"
+#include "libnetlink.h"
+#include "xiart.h"
```

Comment:
I don't think most of these included headers are necessary -- this patch should only include those that are actually needed for the code to compile.

Reply:
Do you want me to remove the unnecessary header files included using " " or also the ones above them?

Reply:
The local header files are all necessary and about the ones above them, xia_socket, fcntl and sys/types are the ones not required.

Reply:
Drop the header files that you've identified that are not required.